src/GithubWebhookJira/Webhook.php

```diff
@@ -0,0 +1,276 @@
+use \JiraRestApi\Issue\IssueService;
+use \JiraRestApi\Issue\Comment;
+use \JiraRestApi\Issue\Transition;
+
+class Webhook {
+
+  /** @var string 'opened' Action */
+  const ACTION_OPENED = 'opened';
+
+  /** @var string 'reopened' Action */
+  const ACTION_REOPENED = 'reopened';
+
+  /** @var string 'edited' Action */
+  const ACTION_EDITED = 'edited';
+
+  /** @var string 'closed' Action */
+  const ACTION_CLOSED = 'closed';
+
+  /** @var int In Progress -> Peer Review Transition */
+  const TRANSITION_PEER_REVIEW = 61;
+
+  /** @var int Peer Review -> Done Transition */
+  const TRANSITION_DONE = 91;
+
+  /** @var int Peer Review -> Review Rejected Transition */
+  const TRANSITION_REVIEW_REJECTED = 121;
+
+  /** @var string Done Resolution */
+  const RESOLUTION_DONE = 'Done';
+
+  /** @var \Silex\Application  Our Silex application */
+  private $_app;
+
+  /** @var \Symfony\Component\HttpFoundation\Request Our Request object */
+  private $_request;
+
+  /** @var \Github\Client Our github client */
+  private $_github;
+
+  /** @var \JiraRestApi\Issue\IssueService Jira cloud issue API service */
+  private $_issue;
+
+  /** @var string The prefix for Jira issues */
+  private $_issue_prefix;
+
+  /** @var string Github Webhook Secret */
+  private $_secret = '';
+
+  /** @var string URL for Jira */
+  private $_jira_url;
+
+  /** @var string Raw data from hook request */
+  private $_raw_data = '';
+
+  /**
+   * Construct this object
+   *
+   * @param \Silex\Application $app
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   */
+  public function __construct(Application $app, Request $request) {
+    $this->_app = $app;
+    $this->_request = $request;
+
+    $this->_secret = getenv('SECRET');
+    $this->_issue_prefix = getenv('JIRA_ISSUE_PREFIX');
+    $this->_jira_url = getenv('JIRA_URL');
+
+    // Setup Github API
+    $this->_github = new Client();
+    $this->_github->authenticate(
+      getenv('GITHUB_API_TOKEN'),
+      Client::AUTH_HTTP_TOKEN
+    );
+
+    // Setup Jira API
+    $config = new ArrayConfiguration([
+      'jiraHost' => $this->_jira_url,
+      'jiraUser' => getenv('JIRA_USERNAME'),
+      'jiraPassword' => getenv('JIRA_PASSWORD'),
+      'jiraLogLevel' => 'EMERGENCY',
+    ]);
+    $this->_issue = new IssueService($config);
+
+    $this->_raw_data = $request->getContent();
+  }
+
+  /**
+   * Check if the webhook is a valid request
+   *
+   * @return boolean
+   */
+  public function isValid() : bool {
+    list($algo, $sig) = explode(
+      '=',
+      $this->_request->headers->get('X-Hub-Signature')
+    );
+    return hash_hmac($algo, $this->_raw_data, $this->_secret) === $sig;
+  }
+
+  /**
+   * Get our data
+   *
+   * @return \stdClass
+   */
+  private function _getData() : \stdClass {
+    return json_decode($this->_raw_data);
+  }
+
+  /**
+   * Process our webhook data
+   *
+   * @return void
+   */
+  public function process() {
+    switch ($this->_getData()->action) {
```

Comment:
this is assuming it's a PR action

Reply:
It should always be a PR action. The webhook only ever sends PRs (per the setup instructions)

Reply:
This is a public repo. Anything can happen.